packages/next-on-pages/src/buildApplication/processVercelFunctions/invalidFunctions.ts

```diff
@@ -71,6 +76,69 @@ async function tryToFixNotFoundRoute(
 	}
 }
 
+/**
+ * Tries to fix potential unnecessary and invalid i18n functions from the Vercel build output.
+ *
+ * This is a workaround for Vercel creating invalid Node.js i18n functions in the build output, and
+ * is achieved by combing through the Vercel build output config to find i18n keys that match the
+ * invalid functions.
+ *
+ * @param collectedFunctions Collected functions from the Vercel build output.
+ * @param opts Options for processing Vercel functions.
+ */
+async function tryToFixI18nFunctions(
+	{ edgeFunctions, invalidFunctions, ignoredFunctions }: CollectedFunctions,
+	{
+		vercelConfig,
+		functionsDir,
+	}: Pick<ProcessVercelFunctionsOpts, 'functionsDir' | 'vercelConfig'>,
+): Promise<void> {
+	if (!invalidFunctions.size || !vercelConfig.routes?.length) {
+		return;
+	}
+
+	const foundI18nKeys = vercelConfig.routes.reduce((acc, route) => {
+		if ('handle' in route) return acc;
+
+		// Matches the format used in certain source route entries in the build output config.
+		// e.g. "src": "/(?<nextLocale>default|en|ja)(/.*|$)"
+		/\(\??<nextLocale>([^)]+)\)/
+			.exec(route.src)?.[1]
+			?.split('|')
+			?.forEach(locale => acc.add(locale));
+
+		return acc;
+	}, new Set<string>());
+
+	if (!foundI18nKeys.size) {
+		// no i18n keys found in the build output config, so we can't fix anything
+		return;
+	}
+
+	for (const [fullPath, fnInfo] of invalidFunctions.entries()) {
+		for (const i18nKey of foundI18nKeys) {
+			const firstRouteSegment = stripFuncExtension(fnInfo.relativePath)
+				.replace(/^\//, '')
+				.split('/')[0];
+
+			if (firstRouteSegment === i18nKey) {
+				const pathWithoutI18nKey = fnInfo.relativePath
+					.replace(new RegExp(`^/${i18nKey}.func`), '/index.func')
+					.replace(new RegExp(`^/${i18nKey}/`), '/');
+				const fullPathWithoutI18nKey = join(functionsDir, pathWithoutI18nKey);
```

Comment:
sorry I am not sure I understand the logic here... aren't you simply stripping the i18n key thus using the non-localized version of the route?

What I mean is that your function takes for example `fr/my-route.func` and changes it to `/my-route.func` so the  route no longer uses the french localization as expected but the default one...

or am I missunderstanding this?

Reply:
that is correct.

something that occurred to me when i was in bed last night is that we should probably add an override for the i18n function's path to the non-i18n function. will add that later today.

Reply:
so... say I have `fr/my-route.func` in french, if your changes apply then we'd get `my-route.func` in say english... so we'd be breaking the internationalization  and you're app would be in a broken state (in which routes that you expect to be in a specific language are not)

am I misunderstanding things?

(because if I am not, I don't think that making the build pass and generate a somewhat wrong application is better than failing and not generating anything at all 🤔)

Reply:
The functions are the same

Reply:
Rather, the internationalisation is all handled internally by the user

Reply:
ah 😮

did you see that somewhere? could you share an example? 🙏